stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -19,12 +19,25 @@ internal class PaymentOptionsViewModel(
     private val eventReporter: EventReporter
 ) : SheetViewModel<PaymentOptionsViewModel.TransitionTarget>(
     config = args.config,
-    prefsRepository = prefsRepository
+    prefsRepository = prefsRepository,
+    previousSelection = args.newCard
 ) {
+    // This field is unique from the selection in sheetViewModel, as the one in sheet view model
+    // is updated any time the add card fragment is in a valid state.  This one will
+    // only be updated when the user presses the Add or selects an item in the carousel.
+    // This will also trigger the closing of the sheet.
     private val _userSelection = MutableLiveData<PaymentSelection>()
     val userSelection: LiveData<PaymentSelection> = _userSelection
 
-    override val newCard = args.newCard
+    // This is used in the case where the last card ws new and not saved.  In this scenario
+    // when the payment options is opened it should jump to the add card, but if the user
+    // presses the back button, they shouldn't transition to it again
+    private var hasTransitionToUnsavedCard = false
+    private val shouldTransitionToUnsavedCard: Boolean
+        get() = !hasTransitionToUnsavedCard &&
+            selection.value != null &&
+            selection.value is PaymentSelection.New &&
+            !(selection.value as PaymentSelection.New).shouldSavePaymentMethod
```

Comment:
I think this can be simplified to:

```
    private val shouldTransitionToUnsavedCard: Boolean
        get() = !hasTransitionToUnsavedCard &&
            (newCardSelection?.let { !it.shouldSavePaymentMethod } ?: false)
```